flow/record/fieldtypes/__init__.py

```diff
@@ -542,23 +542,64 @@ def _unpack(cls, data):
         return data
 
 
+def _is_posixlike_path(path: Any):
+    return isinstance(path, pathlib.PurePath) and path._flavour.sep != "\\" and path._flavour.altsep != "\\"
+
+
+def _is_windowslike_path(path: Any):
+    return isinstance(path, pathlib.PurePath) and (path._flavour.sep == "\\" or path._flavour.altsep == "\\")
```

Comment:
```suggestion
    return isinstance(path, pathlib.PurePath) and "\\" in (path._flavour.sep, path._flavour.altsep)
```

Reply:
Note that this might be slower.

Reply:
![image](https://user-images.githubusercontent.com/110607403/234297303-1b50b5e8-2701-4798-a933-665842c5bfff.png)

Does not seem to be so, at least in my simplified test case. But even if it is/was then we are coding against implementation details which might change from time to time, depending on internal Python code. Anyway, if you do not want the change it's okay, I just thought it would be a bit cleaner not to repeat the string inside that statement. But the benefits are quite limited either way.